prism_compile.c

```diff
@@ -3678,6 +3678,12 @@ pm_compile_node(rb_iseq_t *iseq, const pm_node_t *node, LINK_ANCHOR *const ret,
                     PM_COMPILE(required_param);
                 }
             }
+
+            // If there's only one required element in the parameters
+            // CRuby needs to recognize it as an ambiguous parameter
+            if (body->type == ISEQ_TYPE_BLOCK && requireds_list->size == 1) {
```

Comment:
I think there are more things we have to check here. I think it's that `required_list->size` is 1 and that there are no other parameters in the list at all.